File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-213846-32199/analysis/own-tests-project/src/main/java/customerimporter/ZipCode.java
package customerimporter;

import java.util.List;
import java.util.function.Function;
import java.util.function.Predicate;
import java.util.function.Supplier;

public abstract class ZipCode {

    protected static final int MINIMUM_POSTAL_NUMBER = 1000;
    protected static final int MAXIMUM_POSTAL_NUMBER = 9999;

    private static final List<ZipCodeType> types = List.of(
        new ZipCodeType(OldZipCode::canCreateFrom, OldZipCode::from),
        new ZipCodeType(NewZipCode::canCreateFrom, NewZipCode::from));

    // instance creation

    public static ZipCode from(String aZipCode) {
        return types.stream()
            .filter(aZipCodeType -> aZipCodeType.canCreateFrom(aZipCode))
            .findFirst()
            .orElseThrow(() -> new RuntimeException(invalidZipCodeTypeErrorDescription()))
            .createFrom(aZipCode);
    }

    // error messages

    // The tests expect this wording for a zip code that is neither old nor new
    public static String invalidZipCodeTypeErrorDescription() {
        return "Invalid identification type";
    }

    // asserting - private

    protected static int assertIsValidPostalNumber(String aPostalNumber, CodeValidator aValidator) {
        return aValidator.assertNumberBetween(aPostalNumber, MINIMUM_POSTAL_NUMBER, MAXIMUM_POSTAL_NUMBER);
    }

    // zip code

    public abstract Object value();

    // testing

    public abstract boolean isNew();

    public abstract boolean isOld();

    // values

    public abstract Object newZipCodeIfNone(Supplier<Object> aNoneBlock);

    public abstract Object oldZipCodeIfNone(Supplier<Object> aNoneBlock);

    // types

    private record ZipCodeType(Predicate<String> condition, Function<String, ZipCode> creator) {

        public boolean canCreateFrom(String aZipCode) {
            return condition.test(aZipCode);
        }

        public ZipCode createFrom(String aZipCode) {
            return creator.apply(aZipCode);
        }
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-202259-53403/workdir/src/customerimporter/zip_code.py
from customerimporter.imported_value import ImportedValue


class ZipCode(ImportedValue):

    # instance creation

    @classmethod
    def for_zip_code(cls, a_zip_code):
        return cls.class_handling(a_zip_code).with_value(a_zip_code)

    # validation

    # the four digits of the old zip code, also inside the new one

    @classmethod
    def assert_is_valid_zip_code_number(cls, a_zip_code_number):
        cls.assert_all_are_digits(a_zip_code_number)
        cls.assert_is_between(int(a_zip_code_number), 1000, 9999)

    # testing

    def is_new(self):
        return False

    def is_old(self):
        return False

    # zip code

    def new_zip_code_if_none(self, a_none_block):
        return a_none_block()

    def old_zip_code_if_none(self, a_none_block):
        return a_none_block()
